.github/workflows/ci.yml

```diff
@@ -59,15 +59,52 @@ jobs:
       run: |
         cabal configure --enable-tests --enable-benchmarks --test-show-details=direct
 
+    - name: Update
+      run: |
+        cabal update
```

Comment:
I recommend instead `cabal-update: true` as argument to the setup action.
(Currently this is the default anyway, so the extra `cabal update` step here will just duplicate work.  If you put `cabal-update: true` explicitly, this help the reader of this workflow to see that you though about doing the update.)